kinder/pkg/cri/containerd/alterhelper.go

```diff
@@ -88,6 +88,13 @@ func PullImages(bc *bits.BuildContext, images []string, targetPath string) error
 		if err := bc.RunInContainer("ctr", "image", "export", path+".tar", image); err != nil {
 			return errors.Wrapf(err, "could not save image %q to path %q", image, targetPath)
 		}
+
+		if err := bc.RunInContainer("ctr", "--namespace=k8s.io", "image", "import", path+".tar", "--no-unpack"); err != nil {
+			return errors.Wrapf(err, "could not import image %q from path %q", image, targetPath)
+		}
+		if err := bc.RunInContainer("rm", path+".tar"); err != nil {
+			return errors.Wrapf(err, "could not delete image %q to path %q", image, targetPath)
+		}
```

Comment:
is this how it works pull, export and then import?

also maybe the note on L76-L77 should be removed now.

Reply:
From what I understood, `ctr pull` is not enough to make the images visible from `crictl images`, and so kubeadm was triggering image pull. 
I got the impression that pull internally was capable of resolving to the pulled image and the actual download was not happening, but this was hard to confirm, so I decided to explicitly import the tar because it results in a cleaner workflow.

Also, by deleting the tars before committing allowed me to reduce the size of images and the size of the running containers